### PR TITLE
Update `from-source.zh-cn.md` & `from-source.en-us.md` - Cross Compile Using Zig

### PR DESCRIPTION
**Pull request by editfund-founder**

### CODE REVIEW

**Comment by editfund-founder** [comment]
## Install cc
1.
`which cc`

2.
```sh
$ gcc
Command 'gcc' not found, but can be installed with:
sudo apt install pentium-builder
sudo apt install gcc            
mx@ef:~/3rd-dev/gitea-npm--1.19.3
$ sudo apt install gcc
[sudo] mx 的密码：
正在读取软件包列表... 完成
正在分析软件包的依赖关系树... 完成
正在读取状态信息... 完成                 
建议安装：
  gcc-multilib autoconf automake libtool flex bison gdb gcc-doc
下列【新】软件包将被安装：
  gcc
升级了 0 个软件包，新安装了 1 个软件包，要卸载 0 个软件包，有 212 个软件包未被升级。
需要下载 5,192 B 的归档。
解压缩后会消耗 46.1 kB 的额外空间。
获取:1 http://deb.debian.org/debian bullseye/main amd64 gcc amd64 4:10.2.1-1 [5,192 B]
已下载 5,192 B，耗时 6秒 (882 B/s)
正在选中未选择的软件包 gcc。
(正在读取数据库 ... 系统当前共安装有 297827 个文件和目录。)
准备解压 .../gcc_4%3a10.2.1-1_amd64.deb  ...
正在解压 gcc (4:10.2.1-1) ...
正在设置 gcc (4:10.2.1-1) ...
正在处理用于 man-db (2.9.4-2) 的触发器 ...
mx@ef:~/3rd-dev/gitea-npm--1.19.3
$ 


```
3.
`which cc`

```sh
mx@ef:~/3rd-dev/gitea-npm--1.19.3
$ which cc
/usr/bin/cc
mx@ef:~/3rd-dev/gitea-npm--1.19.3
$ which gcc
/usr/bin/gcc
mx@ef:~/3rd-dev/gitea-npm--1.19.3
$ 

```

**Review by lunny — APPROVED** [review]

File: docs/content/doc/installation/from-source.en-us.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-date: "2016-12-01T16:00:00+02:00"
+date: "2023-06-11T16:00:00+02:00"
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]: I think it's the time of creation, not the time of update.

**Review by wxiaoguang — APPROVED** [review]
Some nits including: 
* The version like "0.10.1" should be general but not hard-coded.
* The "mingw/make" part could share a general document
* The Linux steps and Windows steps are pretty similar 
* `❓ I don't have time to test it myself, if there is any exception, welcome to fix it` it should be removed

Meanwhile it's not bad to have such "zig cross compile" being document.

File: docs/content/doc/installation/from-source.en-us.md [hunk 2]
```diff
@@ -206,3 +206,86 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
...
+- To install:
+```sh
+sudo rm -rf /usr/local/zig && tar -C /usr/local -xf zig-linux-x86_64-0.10.1.tar.xz
+```
+- Environment variable: `export PATH=$PATH:/usr/local/zig-linux-x86_64-0.10.1/`
+- Install cc
+```
+sudo apt install gcc
+which cc
+```
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]: Maybe
```suggestion
Follow [Getting Started of Zig](https://ziglang.org/learn/getting-started/#installing-zig) to install zig.
```
is enough?

I don't think it's appropriate to introduce too much about "how to install zig" into Gitea's documentation.

File: docs/content/doc/installation/from-source.en-us.md [hunk 3]
```diff
@@ -206,3 +206,86 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
...
+CC="zig cc -target x86_64-linux-gnu" \
+CGO_ENABLED=1 \
+CGO_CFLAGS="-O2 -g -pthread" \
+CGO_LDFLAGS="-linkmode=external -v"
+GOOS=linux \
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+❓ I don't have time to test it myself, if there is any exception, welcome to fix it
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]: TBH, I don't think they are appropriate:
```
✅Proven success
✅Proven success (Gitea version 1.20.0-rc0,compiled 96.4M, select SQLite3 can be installed normally)
❓ I don't have time to test it myself, if there is any exception, welcome to fix it
```

It doesn't look like the official documentation for Gitea, it looks like a personal article ”How to compile Gitea with zig“. How about removing them? If someone get a exception, they will submit an issue.

File: docs/content/doc/installation/from-source.en-us.md [hunk 4]
```diff
@@ -206,3 +206,86 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
...
+
+Make `GIT BASH` support make:
+- Go to https://sourceforge.net/projects/ezwinports/files/ and download `make-4.4.1-without-guile-w32-bin.zip`
+- Decompress the file
+- Copy all the extracted files to the git installation directory: . \Program Files\Git\mingw64\, merge the folders, and choose not to replace the files that pop up if they need to be replaced.
+- So that you can execute make in the git bash window (the compiler still needs to be installed separately when compiling)
+
+To make `GIT BASH` support make (second method):
+- Download: [https://github.com/niXman/mingw-builds-binaries](https://github.com/niXman/mingw-builds-binaries)
+- Copy mingw32-make.exe to `C:\Program Files\Git\mingw64\bin`, renamed make.exe
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]: Ditto.

**Review by wolfogre — COMMENTED** [review]
Thank you for this PR, it's awesome to complie Gitea with zig. And I have some suggestions about tone.

File: docs/content/doc/installation/from-source.en-us.md (hunk 4 above)
Comment (editfund-founder) [reply docs/content/doc/installation/from-source.en-us.md]: 💕Thanks

File: docs/content/doc/installation/from-source.en-us.md (hunk 2 above)
Comment (editfund-founder) [reply docs/content/doc/installation/from-source.en-us.md]: 💕Thanks

File: docs/content/doc/installation/from-source.en-us.md (hunk 3 above)
Comment (editfund-founder) [reply docs/content/doc/installation/from-source.en-us.md]: 💕I love the community, I love you.

File: docs/content/doc/installation/from-source.en-us.md [hunk 5]
```diff
@@ -206,3 +206,57 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
...
+CGO_CFLAGS="-O2 -g -pthread" \
+CGO_LDFLAGS="-linkmode=external -v"
+GOOS=linux \
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+2. cross-compile (Linux ➝ Windows)
+```sh
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]:
````suggestion
2. Cross-compile (Linux ➝ Windows)

```sh
````

File: docs/content/doc/installation/from-source.en-us.md [hunk 6]
```diff
@@ -206,3 +206,57 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
...
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+## Compile or cross-compile with Zig using Windows
+
+Compile with `GIT BASH`.
+
+1. Compile (Windows ➝ Windows)
+```
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]:
````suggestion
1. Compile (Windows ➝ Windows)

```sh
````

File: docs/content/doc/installation/from-source.en-us.md [hunk 7]
```diff
@@ -206,3 +206,57 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
...
+CC="zig cc -target x86_64-windows-gnu" \
+CGO_ENABLED=1 \
+CGO_CFLAGS="-O2 -g -pthread" \
+GOOS=windows \
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+2. Cross-compile (Windows ➝ Linux)
+```
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]:
````suggestion
```

2. Cross-compile (Windows ➝ Linux)

```sh
````

File: docs/content/doc/installation/from-source.zh-cn.md [hunk 8]
```diff
@@ -103,3 +103,58 @@ GOOS=linux GOARCH=arm64 make build
...
+CGO_CFLAGS="-O2 -g -pthread" \
+CGO_LDFLAGS="-linkmode=external -v"
+GOOS=linux \
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+2.交叉编译(Linux ➝ Windows)
+```sh
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.zh-cn.md]:
````suggestion
2.交叉编译(Linux ➝ Windows)

```sh
````

File: docs/content/doc/installation/from-source.zh-cn.md [hunk 9]
```diff
@@ -103,3 +103,58 @@ GOOS=linux GOARCH=arm64 make build
...
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+## 使用Windows与Zig编译或交叉编译
+
+使用`GIT BASH`编译。
+
+1.编译(Windows ➝ Windows)
+```
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.zh-cn.md]:
````suggestion
1.编译(Windows ➝ Windows)

```sh
````

File: docs/content/doc/installation/from-source.zh-cn.md [hunk 10]
```diff
@@ -103,3 +103,58 @@ GOOS=linux GOARCH=arm64 make build
...
+CGO_ENABLED=1 \
+CGO_CFLAGS="-O2 -g -pthread" \
+GOOS=windows \
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+2.交叉编译(Windows ➝ Linux)
+```
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.zh-cn.md]:
````suggestion
2.交叉编译(Windows ➝ Linux)

```sh
````

**Review by wolfogre — APPROVED** [review]

File: docs/content/doc/installation/from-source.en-us.md [hunk 11]
```diff
@@ -206,3 +206,61 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
 `.zshrc`.
 
 YMMV and these scripts may need further improvement.
+
+## Compile or cross-compile using Linux with Zig
+
+Follow [Getting Started of Zig](https://ziglang.org/learn/getting-started/#installing-zig) to install zig.
+
+1. Compile (Linux ➝ Linux)
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]:
```suggestion
- Compile (Linux ➝ Linux)
```

File: docs/content/doc/installation/from-source.en-us.md [hunk 12]
```diff
@@ -206,3 +206,61 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
...
+CGO_ENABLED=1 \
+CGO_CFLAGS="-O2 -g -pthread" \
+CGO_LDFLAGS="-linkmode=external -v"
+GOOS=linux \
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+2. Cross-compile (Linux ➝ Windows)
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]:
```suggestion
- Cross-compile (Linux ➝ Windows)
```

File: docs/content/doc/installation/from-source.en-us.md [hunk 13]
```diff
@@ -206,3 +206,61 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
...
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+## Compile or cross-compile with Zig using Windows
+
+Compile with `GIT BASH`.
+
+1. Compile (Windows ➝ Windows)
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]:
```suggestion
- Compile (Windows ➝ Windows)
```

File: docs/content/doc/installation/from-source.en-us.md [hunk 14]
```diff
@@ -206,3 +206,61 @@ Similarly, a script for zsh-completion can be found at [`contrib/autocompletion/
...
+CC="zig cc -target x86_64-windows-gnu" \
+CGO_ENABLED=1 \
+CGO_CFLAGS="-O2 -g -pthread" \
+GOOS=windows \
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+2. Cross-compile (Windows ➝ Linux)
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.en-us.md]:
```suggestion
- Cross-compile (Windows ➝ Linux)
```

File: docs/content/doc/installation/from-source.zh-cn.md [hunk 15]
```diff
@@ -103,3 +103,61 @@ GOOS=linux GOARCH=arm64 make build
 ```bash
 CC=aarch64-unknown-linux-gnu-gcc GOOS=linux GOARCH=arm64 TAGS="bindata sqlite sqlite_unlock_notify" make build
 ```
+
+## 使用Linux与Zig编译或交叉编译
+
+按照[Getting Started of Zig](https://ziglang.org/learn/getting-started/#installing-zig)来安装zig。
+
+1.编译(Linux ➝ Linux)
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.zh-cn.md]:
```suggestion
- 编译(Linux ➝ Linux)
```

File: docs/content/doc/installation/from-source.zh-cn.md [hunk 16]
```diff
@@ -103,3 +103,61 @@ GOOS=linux GOARCH=arm64 make build
...
+CGO_ENABLED=1 \
+CGO_CFLAGS="-O2 -g -pthread" \
+CGO_LDFLAGS="-linkmode=external -v"
+GOOS=linux \
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+2.交叉编译(Linux ➝ Windows)
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.zh-cn.md]:
```suggestion
- 交叉编译(Linux ➝ Windows)
```

File: docs/content/doc/installation/from-source.zh-cn.md [hunk 17]
```diff
@@ -103,3 +103,61 @@ GOOS=linux GOARCH=arm64 make build
...
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+## 使用Windows与Zig编译或交叉编译
+
+使用`GIT BASH`编译。
+
+1.编译(Windows ➝ Windows)
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.zh-cn.md]:
```suggestion
- 编译(Windows ➝ Windows)
```

File: docs/content/doc/installation/from-source.zh-cn.md [hunk 18]
```diff
@@ -103,3 +103,61 @@ GOOS=linux GOARCH=arm64 make build
...
+CC="zig cc -target x86_64-windows-gnu" \
+CGO_ENABLED=1 \
+CGO_CFLAGS="-O2 -g -pthread" \
+GOOS=windows \
+GOARCH=amd64 \
+TAGS="bindata sqlite sqlite_unlock_notify" \
+make build
+```
+
+2.交叉编译(Windows ➝ Linux)
```
Comment (wolfogre) [inline docs/content/doc/installation/from-source.zh-cn.md]:
```suggestion
- 交叉编译(Windows ➝ Linux)
```

**Pull request merged**